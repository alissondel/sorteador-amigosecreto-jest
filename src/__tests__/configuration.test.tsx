import { render } from "@testing-library/react"
import { RecoilRoot } from "recoil"
import ConfigurationPage from "../pages/configuration"

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate
  }
})

describe('A pagina de configuração', () => {
  it('Deve ser reenderizada corretamente', () => {
    const container = render(
      <RecoilRoot>
        <ConfigurationPage />
      </RecoilRoot>
    )

    // Teste de SnapShoot
    expect(container).toMatchSnapshot()
  })
})