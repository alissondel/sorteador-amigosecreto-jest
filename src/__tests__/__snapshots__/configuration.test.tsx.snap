// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A pagina de configuração Deve ser reenderizada corretamente 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex items-center justify-center min-h-screen bg-zinc-500"
      >
        <div
          class="flex flex-col rounded-md shadow-sm transition-shadow py-4 px-8"
        >
          <header
            class="flex flex-col justify-between items-center bg-zinc-50"
          >
            <div
              class="flex flex-row items-center mb-2"
            >
              <img
                alt="Logo"
                class="h-50 w-50 bg-blue-500"
                src="/images/logo.png"
              />
            </div>
            <h2
              class="text-3xl font-semibold"
            >
              Vamos Começar
            </h2>
          </header>
          <form>
            <div
              class="flex flex-col items-center bg-zinc-50 w-full mt-4"
            >
              <input
                class="shadow appearance-none border border-red-500 rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
                placeholder="Inserir nomes dos participantes"
                type="text"
                value=""
              />
              <button
                class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline w-full disabled:bg-gray-500
          cursor-not-allowed  
          "
                disabled=""
                type="submit"
              >
                Adicionar
              </button>
            </div>
          </form>
          <p
            class="text-center text-lg font-semibold text-zinc-950"
          >
            Não há participantes no momento...
          </p>
          <footer
            class="bg-zinc-50 w-full mt-4"
          >
            <button
              class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline w-full disabled:bg-gray-500
          cursor-not-allowed  
          "
              disabled=""
            >
              Iniciar Brincadeira
            </button>
          </footer>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex items-center justify-center min-h-screen bg-zinc-500"
    >
      <div
        class="flex flex-col rounded-md shadow-sm transition-shadow py-4 px-8"
      >
        <header
          class="flex flex-col justify-between items-center bg-zinc-50"
        >
          <div
            class="flex flex-row items-center mb-2"
          >
            <img
              alt="Logo"
              class="h-50 w-50 bg-blue-500"
              src="/images/logo.png"
            />
          </div>
          <h2
            class="text-3xl font-semibold"
          >
            Vamos Começar
          </h2>
        </header>
        <form>
          <div
            class="flex flex-col items-center bg-zinc-50 w-full mt-4"
          >
            <input
              class="shadow appearance-none border border-red-500 rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
              placeholder="Inserir nomes dos participantes"
              type="text"
              value=""
            />
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline w-full disabled:bg-gray-500
          cursor-not-allowed  
          "
              disabled=""
              type="submit"
            >
              Adicionar
            </button>
          </div>
        </form>
        <p
          class="text-center text-lg font-semibold text-zinc-950"
        >
          Não há participantes no momento...
        </p>
        <footer
          class="bg-zinc-50 w-full mt-4"
        >
          <button
            class="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline w-full disabled:bg-gray-500
          cursor-not-allowed  
          "
            disabled=""
          >
            Iniciar Brincadeira
          </button>
        </footer>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
